dive_core/command_hierarchy.cpp

```diff
@@ -1929,12 +1886,11 @@ void CommandHierarchyCreator::AppendPacketFieldNodes(const IMemoryManager &mem_m
             }
 
             uint32_t dword_value = 0;
-            uint64_t dword_va_addr = va_addr + field_dword * sizeof(uint32_t);
-            bool     ret = mem_manager.CopyMemory(&dword_value,
-                                              submit_index,
-                                              dword_va_addr,
-                                              sizeof(uint32_t));
-            DIVE_VERIFY(ret);  // This should never fail!
+            // (field_dword - 1) since each field is always 1 32bit register, we don't have any
+            // 64bit field
+            uint64_t dword_va_addr = va_addr + (field_dword - 1) * sizeof(uint32_t);
```

Comment:
@shanminchao could you confirm if this is the right fix? m_dword seems to be the number of dwords that including current field whereas here we want to have the offset of current field.

Reply:
I don't understand what is being fixed here. Before, va_addr pointed to the beginning of the packet, but this PR offsets it by the size of the 32-bit header, and then it's further offseted backwards here with a -1. Was there a specific issue being fixed here?

Reply:
packet_field.m_dword is supposed to indicate which dword the field is in. I forget if the dword # starts at 0 or 1. But in any case, it was used to calculate an offset

Reply:
currently, it starts from 1, so if i understand correctly, it is not offset, it is the dword count that includes current field, maybe this is not intended? 
https://github.com/google/dive/blob/e3b9bd69ade10131118f0a482c2988bbc704f843/dive_core/generateAdrenoHeader.py#L133

Reply:
It's a count, used to calculate the offset.
So for the 1st dword field after the header, you'd calculate the offset by multiplying m_dword with sizeof(uint32_t). It is a count, but a count used to calculate an offset. Hence the original code:

uint32_t dword_va_addr = va_addr + field_dword * sizeof(uint32_t)

Reply:
the reason for the fix is because `AppendPacketFieldNodes` is called in different places. the place where it is called in the lambda function doesn't have the PM4 packets, to be consistent, i made va_addr does not contain the PM4 packets. (this is why the previous change offset size of 32-bit header) And since va_addr doesn't contain the header 32bit, there is this (field_dword - 1) here

Reply:
it is confusing, but i could put better comments to explain it better. or do you suggest to fix it in a different way?

Reply:
Ooohhh I see. So the lambda function never worked before? I could've sworn I tested it! Or was it buggy only in the context of bindless?

So it's a result of a bug fix. I don't think the explanation is lacking; I just didn't understand what you were trying to do, having not noticed the other call site for this function.

Reply:
I notice the bug in the bindless case when i was checking the content of the descriptors.

Reply:
Got it, thanks for fixing!